CMakeLists.txt

```
@@ -105,6 +105,12 @@ target_link_libraries(stdexec INTERFACE Threads::Threads)
 # Use C++20 standard
 target_compile_features(stdexec INTERFACE cxx_std_20)
 
+# Enable GPU compilation when using NVHPC compiler
+if (CMAKE_CXX_COMPILER_ID STREQUAL "NVHPC")
+target_compile_features(stdexec INTERFACE -stdpar=gpu)
+endif()
```

Maybe you want something like this? I'm not a CMake expert, but I copy/paste very well.

```suggestion
target_compile_options(stdexec INTERFACE
                       $<$<CXX_COMPILER_ID:NVHPC>:-stdpar=gpu>
                       )
```

---

This is beyond me. This is the kind of thing where I would have have to ask @trxcllnt or @robertmaynard

---

Almost correct!

You want:
```cmake
target_compile_options(stdexec INTERFACE
                       $<$<COMPILE_LANG_AND_ID:CXX,NVHPC>:-stdpar=gpu>
                       )
target_link_options(stdexec INTERFACE
                       $<$<LINK_LANG_AND_ID:CXX,NVHPC>:-stdpar=gpu>
                       )                       
```

We use `COMPILE_LANG_AND_ID` since we only want `-stdpar=gpu` provided to C++ sources ( not fortran , C, or CUDA ). I believe we also need to pass the `-stdpar=gpu` to the linker as well, so we need `target_link_options` for that.

---

```suggestion
target_compile_options(stdexec INTERFACE
                       $<$<COMPILE_LANG_AND_ID:CXX,NVHPC>:-stdpar=gpu>
                       )
target_link_options(stdexec INTERFACE
                       $<$<LINK_LANG_AND_ID:CXX,NVHPC>:-stdpar=gpu>
                       )
```

---

Hm, something about these changes seem to have broken the build for nvc++: https://github.com/NVIDIA/stdexec/actions/runs/3660473250/jobs/6187644769

---

Error is:
```
2022-12-09T20:48:16.2261936Z nvc++-Fatal-The option -stdpar=gpu,multicore is not yet supported by nvc++
```

So some tests are already building with `-stdpar=multicore` and now that is being combined with `stdpar=gpu`.